internal/engines/envsubst.go

```diff
@@ -0,0 +1,63 @@
+package engines
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/a8m/envsubst"
+)
+
+type EnvsubstEngine struct{}
+
+func (e *EnvsubstEngine) RenderFile(file string, w io.Writer, data map[string]any) error {
+	buf, err := os.ReadFile(file)
+	if err != nil {
+		return err
+	}
+
+	err = envsubstRender(w, buf, data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *EnvsubstEngine) Render(r io.Reader, w io.Writer, data map[string]any) error {
+	buf, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+
+	err = envsubstRender(w, buf, data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func envsubstRender(w io.Writer, input []byte, data map[string]any) error {
```

Comment:
this function is not needed. can place all the logic inside `Render` and call `Render` from `RenderFile` with the file as the reader